app/uk/gov/hmrc/nationalinsurancedesstub/controllers/NationalInsuranceSummaryController.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.nationalinsurancedesstub.controllers
+
+import play.api.libs.json.Json
+import play.api.mvc._
+import uk.gov.hmrc.nationalinsurancedesstub.common.StubResource
+import uk.gov.hmrc.nationalinsurancedesstub.models.NationalInsuranceSummary
+import uk.gov.hmrc.nationalinsurancedesstub.repositories.NationalInsuranceSummaryRepository
+import uk.gov.hmrc.play.microservice.controller.BaseController
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+trait NationalInsuranceSummaryController extends BaseController with StubResource {
+  val repository: NationalInsuranceSummaryRepository
+
+  def fetch(utr: String, taxYear: String) = Action.async { implicit request =>
```

Comment:
Could perhaps do the same as I did and move the futures out of the for/comp into vals so that they can be processed i parallel?

```
  final def find(utr: String, taxYear: String) = Action async {
    val annualSummaryDocumentFuture = findResource("/resources/individual-benefits/happy_path_annual_summary.json")
    val individualBenefitsFuture = individualBenefitsRepository.fetch(utr, taxYear)

    (for {
      annualSummaryDocument <- annualSummaryDocumentFuture
      individualBenefitsOption <- individualBenefitsFuture
    } yield individualBenefitsOption match {
      case Some(_) => Ok(annualSummaryDocument)
      case None => NotFound
    }) recover {
      case _ => InternalServerError
    }
  }
```